config/postgresql/config.go

```diff
@@ -245,6 +245,10 @@ func Configure(p *config.Provider) {
 				Type:      "Server",
 				Extractor: rconfig.ExtractResourceIDFuncPath,
 			},
+			"key_vault_key_id": config.Reference{
```

Comment:
Oh, sorry :) I mean the IDs for the `azurerm_postgresql_server_key` resources. I wonder whether we can use `NameAsIdentifier` for this resource.

Reply:
@ulucinar got it:) looks like
```
/subscriptions/038f2b7c-3265-43b8-8624-c9ad5da610a8/resourceGroups/postgresql-server-data-encryption-example/providers/Microsoft.DBforPostgreSQL/servers/postgresql-server-data-encryption-example/keys/data-encryption_data-encryption_51a88b3caf68453c9ccd5aab76582f56
```
so there is dynamic part at the end

Reply:
The last token (`51a88b3caf68453c9ccd5aab76582f56`) looks like something Azure-generated, what do you think?

Reply:
I think we are fine here. If you'd like to give it a try, having the following configuration:
```go
		r.ExternalName = config.IdentifierFromProvider
		r.ExternalName.GetExternalNameFn = common.GetNameFromFullyQualifiedID
		r.ExternalName.GetIDFn = func(_ context.Context, externalName string, parameters map[string]interface{}, _ map[string]interface{}) (string, error) {
			return fmt.Sprintf("%s/keys/%s", parameters["server_id"], externalName), nil
		}
```
we will have just the last component of the ID as the external name.

Reply:
Hi @ytsarev,
I think the error message is coming from the native provider because we are passing an empty external-name with `config.IdentifierFromProvider`.  If you'd like, could you please try with the following config:
```go
		r.ExternalName = config.NameAsIdentifier
		r.ExternalName.SetIdentifierArgumentFn = config.NopSetIdentifierArgument
		r.ExternalName.GetExternalNameFn = common.GetNameFromFullyQualifiedID
		r.ExternalName.GetIDFn = func(_ context.Context, externalName string, parameters map[string]interface{}, _ map[string]interface{}) (string, error) {
			return fmt.Sprintf("%s/keys/%s", parameters["server_id"], externalName), nil
		}
```
Also I think we are missing an example manifest for the new resource. Could you please add the manifest you're using as an example?

Reply:
@ulucinar great, will try this one and thanks a lot for the example catch 👍

Reply:
@ulucinar looks like it is still not getting populated

```
Events:
  Type     Reason                         Age                   From                                                                      Message
  ----     ------                         ----                  ----                                                                      -------
  Warning  CannotObserveExternalResource  51s (x11 over 6m53s)  managed/dbforpostgresql.azure.jet.crossplane.io/v1alpha2, kind=serverkey  cannot run refresh: refresh failed: The number of path segments is not divisible by 2 in "subscriptions/038f2b7c-3265-43b8-8624-c9ad5da610a8/resourceGroups/postgresql-server-data-encryption-example/providers/Microsoft.DBforPostgreSQL/servers/postgresql-server-data-encryption-example/keys": : File name: main.tf.json
```

I've added the example to the PR, thanks a ton for the catch

Reply:
Thank you @ytsarev! What was missing is a run of `make generate` to inject the initializer.